### ISSUE TITLE
Weird Paramspec kwargs unpack behavior

### ISSUE THREAD

**Issue by hmc-cs-mdrissi** [microsoft/pyright#3517]
**Describe the bug**
**kwargs on a type annotated as P.kwargs leads inferred type to be weird and miss errors.

```python
from typing import Generic

from typing_extensions import ParamSpec

P = ParamSpec("P")


class Serializer(Generic[P]):
    def serialize_arguments(self, *args: P.args, **kwargs: P.kwargs) -> dict[str, str]:
        # In real code **{} is some actual other dict, but simplified here.
        params: dict[str, str] = {**{}, **kwargs} # revealed type of params is dict[Unknown | str, Unknown | object]
        return params
```

Interestingly if you replace {**{}, **kwargs}  with {**kwargs} it does give a correct type error.

**To Reproduce**
Run pyright and see 0 errors.

**Expected behavior**
I would expect an error on params: dict[str, str] line as there's nothing ensuring P.kwargs is string valued.

**VS Code extension or command-line**
1.1.249. Doing a bisect this behavior changed around 1.1.229. Before that version this did report 1 error `Expression of type "dict[Unknown | str, Unknown | object]" cannot be assigned to declared type "dict[str, str]"`. But starting from that version the error disappeared.

This is very minor and just a weird thing I discovered when tweaking some paramspec heavy code.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.250, which I just published. It will also be included in the next release of pylance.
